src/ktopt_planning_context.cpp

```diff
@@ -177,6 +180,42 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   // TODO: These should be parameters
   trajopt.AddDurationConstraint(0.5, 5);
 
+  // process path_constraints
+  if (!req.path_constraints.position_constraints.empty())
+  {
+    for (const auto& position_constraint : req.path_constraints.position_constraints)
+    {
+      // Extract the bounding box's center (primitive pose)
+      const auto& primitive_pose = position_constraint.constraint_region.primitive_poses[0];
+      Eigen::Vector3d box_center(primitive_pose.position.x, primitive_pose.position.y, primitive_pose.position.z);
+
+      // Extract dimensions of the bounding box from
+      // constraint_region.primitives Assuming it is a box
+      // (shape_msgs::SolidPrimitive::BOX) and has dimensions in [x, y, z]
+      const auto& primitive = position_constraint.constraint_region.primitives[0];
+      if (primitive.type == shape_msgs::msg::SolidPrimitive::BOX)
+      {
+        double x_dim = primitive.dimensions[0] / 2.0;
+        double y_dim = primitive.dimensions[1] / 2.0;
+        double z_dim = primitive.dimensions[2] / 2.0;
+
+        // Calculate the lower and upper bounds based on the box dimensions
+        // around the center
+        Eigen::Vector3d lower_bound = box_center - Eigen::Vector3d(x_dim, y_dim, z_dim);
+        Eigen::Vector3d upper_bound = box_center + Eigen::Vector3d(x_dim, y_dim, z_dim);
+
+        // Add position constraint to each knot point of the trajectory
+        for (int i = 0; i < 10; ++i)
+        {
+          trajopt.AddPathPositionConstraint(
+              std::make_shared<PositionConstraint>(&plant, plant.world_frame(), lower_bound, upper_bound, link_ee_frame,
+                                                   Eigen::Vector3d(0.0, 0.1, 0.0), &plant_context),
+              static_cast<double>(i) / 10.0);
```

Comment:
The `10` here should not be hard-coded; it should be a parameter. See e.g. how the collision constraints are setup with `num_collision_check_points`:

https://github.com/moveit/moveit_drake/blob/main/res/ktopt_moveit_parameters.yaml#L31